model/document.js

```diff
@@ -197,7 +197,32 @@ module.exports = memoize(function (db) {
 			if (statusLog.length) data.statusLog = statusLog;
 
 			return data;
-		} }
+		} },
+		toWSSchema: {
+			value: function (ignore) {
+				var schema = {
+					type: "array",
+					items: {
+						type: "object",
+						properties: {
+							code: {
+								type: "enum",
+								ref: "documents"
+							},
+							data: {}
+						}
+					}
+				};
+				schema.items.properties.files =
+					this.files.map._descriptorPrototype_.type.prototype.toWSSchema();
+				schema.items.properties.owner = { type: "string" };
+				if (this.dataForm.constructor !== this.database.FormSectionBase) {
+					schema.items.properties.data = this.dataForm.toWSSchema();
+					schema.items.dataForms = [ { title: "", properties: [] } ];
```

Comment:
@raunokulla 
this property should allow consumers to split the properties of the interesting entities in dataForms, as per our view model. 
I update this gist (manually..) with example
https://gist.github.com/roxanadina/c7b158a0336189845340ad667f7e7911#file-schema-els-json-L270-L279

while in nested entities, you should provide this prop at that specific level definition
https://gist.github.com/roxanadina/c7b158a0336189845340ad667f7e7911#file-schema-els-json-L315-L318

It should be more or less the same thing as we have in our view model, but respect the structure and properties of the schema.
These properties are right now only for project @mtuchowski is working on, so feel free to also ask him any questions